p22/src/figures.rs

```diff
@@ -0,0 +1,57 @@
+pub struct Point {
+    x: f32,
+    y: f32,
+}
+
+pub struct Circle {
+    center: Point,
+    radius: f32,
+}
+
+pub struct Triangle {
+    p1: Point,
+    p2: Point,
+    p3: Point,
+}
+
+pub struct Rectangle {
+    p1: Point,
+    p2: Point,
+}
+
+pub enum Shape {
+    Point(Point),
+    Circle(Circle),
+    Triangle(Triangle),
+    Rectangle(Rectangle),
+}
+
+/// Compute area of shape
+pub fn area(shape: Shape) -> f32 {
+    match shape {
+        Shape::Point(_) => 0.0,
+        Shape::Circle(Circle { center, radius }) => std::f32::consts::PI * radius * radius,
+        Shape::Triangle(Triangle { p1, p2, p3 }) => {
+            (p1.x * (p2.y - p3.y) + p2.x * (p3.y - p1.y) + p3.x * (p1.y - p2.y)).abs() / 2.0
```

Comment:
The problem description it's asked to implement separate functions for all shape types, i.e. you need `area_point`, `area_circle`, etc. You then can reuse those functions in implementation of `area_shape`.